<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PassphraseGenerationDialog</class>
 <widget class="QDialog" name="PassphraseGenerationDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>498</width>
    <height>571</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>465</width>
    <height>457</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Generate Passphrase</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="2" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="9" column="2">
      <widget class="QSpinBox" name="uiCustomSymbolsMax">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>99</number>
       </property>
      </widget>
     </item>
     <item row="8" column="0" colspan="2">
      <widget class="QCheckBox" name="uiCustomSymbols">
       <property name="text">
        <string>Custom:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="QCheckBox" name="uiLowerAtoZ">
       <property name="text">
        <string>Lowercase a-z</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="2">
      <widget class="QCheckBox" name="uiNumbers">
       <property name="text">
        <string>Numbers 0-9</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="2">
      <widget class="QCheckBox" name="uiSymbols">
       <property name="text">
        <string>Symbols:</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>` ~ ! @ # $ % ^ &amp; * ( ) _ - + = { } [ ]  | : ; &quot; ' &lt; &gt; , . ? /</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QRadioButton" name="uiLength8">
         <property name="text">
          <string>8</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="uiLength12">
         <property name="text">
          <string>12</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="uiLength16">
         <property name="text">
          <string>16</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="uiCustomLength">
         <property name="text">
          <string>Custom</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="uiCustomLengthValue">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximum">
          <number>128</number>
         </property>
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="3" column="2">
      <widget class="QSpinBox" name="uiLowerAtoZMax">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>99</number>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="QCheckBox" name="uiUpperAtoZ">
       <property name="text">
        <string>Uppercase A-Z</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QSpinBox" name="uiNumbersMax">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>99</number>
       </property>
      </widget>
     </item>
     <item row="12" column="0" colspan="3">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QComboBox" name="uiPassphraseCount">
         <property name="currentText">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="uiRegenerate">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>&amp;Regenerate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="uiInsertAll">
         <property name="text">
          <string>Insert &amp;all</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="9" column="1">
      <widget class="QLineEdit" name="uiCustomSymbolsValue">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QSpinBox" name="uiUpperAtoZMax">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>99</number>
       </property>
      </widget>
     </item>
     <item row="7" column="2">
      <widget class="QSpinBox" name="uiSymbolsMax">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>99</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>&amp;Length:</string>
       </property>
       <property name="buddy">
        <cstring>uiLength8</cstring>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Select &amp;passphrase:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="buddy">
        <cstring>uiPassphrases</cstring>
       </property>
      </widget>
     </item>
     <item row="11" column="0" colspan="3">
      <widget class="QListWidget" name="uiPassphrases"/>
     </item>
     <item row="1" column="0" colspan="3">
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="1" colspan="2">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Limit group occurrence (1 to max chars)</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Groups</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
  <action name="actionRegenerate">
   <property name="text">
    <string>Regenerate</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>uiLength8</tabstop>
  <tabstop>uiLength12</tabstop>
  <tabstop>uiLength16</tabstop>
  <tabstop>uiCustomLength</tabstop>
  <tabstop>uiCustomLengthValue</tabstop>
  <tabstop>uiLowerAtoZ</tabstop>
  <tabstop>uiLowerAtoZMax</tabstop>
  <tabstop>uiUpperAtoZ</tabstop>
  <tabstop>uiUpperAtoZMax</tabstop>
  <tabstop>uiNumbers</tabstop>
  <tabstop>uiNumbersMax</tabstop>
  <tabstop>uiSymbols</tabstop>
  <tabstop>uiSymbolsMax</tabstop>
  <tabstop>uiCustomSymbols</tabstop>
  <tabstop>uiCustomSymbolsValue</tabstop>
  <tabstop>uiCustomSymbolsMax</tabstop>
  <tabstop>uiPassphrases</tabstop>
  <tabstop>uiPassphraseCount</tabstop>
  <tabstop>uiRegenerate</tabstop>
  <tabstop>uiInsertAll</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>uiCustomLengthValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PassphraseGenerationDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomLength</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiCustomLengthValue</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiLowerAtoZ</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiLowerAtoZMax</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiUpperAtoZ</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiUpperAtoZMax</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiNumbers</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiNumbersMax</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiSymbols</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiSymbolsMax</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomSymbols</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiCustomSymbolsValue</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomSymbols</sender>
   <signal>toggled(bool)</signal>
   <receiver>uiCustomSymbolsMax</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomLength</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomSymbols</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomSymbolsMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>427</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiCustomSymbolsValue</sender>
   <signal>textChanged(QString)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiLength12</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiLength8</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiLength16</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiLowerAtoZ</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiLowerAtoZMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>427</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiNumbers</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiNumbersMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>427</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiSymbolsMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>427</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiSymbols</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiRegenerate</sender>
   <signal>clicked()</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>405</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiUpperAtoZ</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiUpperAtoZMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>427</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PassphraseGenerationDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>436</y>
    </hint>
    <hint type="destinationlabel">
     <x>232</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiPassphraseCount</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionRegenerate</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>404</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiPassphrases</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>PassphraseGenerationDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>232</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
